### PR TITLE
fix(components): input-number 0 can not triger change event

### PR DESCRIPTION
**Pull request by StephenKe**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/input-number/src/input-number.vue [hunk 1]
```diff
@@ -275,7 +275,7 @@ watch(
   (value) => {
     const userInput = verifyValue(data.userInput)
     const newValue = verifyValue(value, true)
-    if (!userInput || userInput !== newValue) {
+    if (userInput !== 0 && (!userInput || userInput !== newValue)) {
```
Comment (jw-foss) [inline packages/components/input-number/src/input-number.vue:278]: @StephenKe maybe we can use `isNumber(userInput)` here for this branch. Something like:

```ts
if (!isNumber(userInput) && !userInput || otherConditions) {}
```

Comment (jw-foss) [reply packages/components/input-number/src/input-number.vue:278]: Plus, since this is a good catch, I'd like to ask you to write a test case for this edge case.

**Comment by jw-foss** [comment]
@StephenKe I left a comment for your consideration, let me know if you need anything. 🏓

**Comment by wldrve** [comment]
Having the same issue! Please merge this fix ASAP!

File: packages/components/input-number/src/input-number.vue (hunk 1 above)
Comment (StephenKe) [reply packages/components/input-number/src/input-number.vue]: > Plus, since this is a good catch, I'd like to ask you to write a test case for this edge case.

Added a edge case in change event.But wrapper.find('input').setValue(0) cannot catch it.I wonder if it should test props.modelValue in watch.

**Comment by StephenKe** [comment]
> @StephenKe I left a comment for your consideration, let me know if you need anything. 🏓

@jw-foss Thanks for the quick reply.Already updated  pr and consideration.Plz review👀

**Comment by wldrve** [comment]
Please merge this before next monday and release it! It is urgent for one of our clients.

**Review by jw-foss — APPROVED** [review]

**Comment by jw-foss** [comment]
@StephenKe changes LGTM. @wldrve we will perform a release ASAP.
